src/assets/js/main.js

```diff
@@ -247,6 +241,31 @@ $(document).ready(function () {
 		}
 	});
 
+	$(document).click(function(event) {
+		//Watch for clicks and check if it is outside the video modal
+		if (!$(event.target).closest(".video-modal-wrapper").length || videoModal.hasClass==active) {
+			closevideo();
+		}
+	  });
+
+	document.onkeydown = function(evt) {
+		// Watch for escape key and close video modal if active
+		evt = evt || window.event;
+		if (evt.keyCode == 27 || videoModal.hasClass==active) {
+			closevideo();
+		}
+	};
+
+	function closevideo(){
+		// pausevideo
+		videoModalPlayPauseButton.children().eq(0).click();
```

Comment:
`videoModalPlayPauseButton` is only initialised when the play/pause button is pressed. Its possible for `closevideo` to be called before `videoModalPlayPauseButton` is initialised, eg refresh the page and immediately hit escape. This causes errors to be logged ("`videoModalPlayPauseButton is not defined`").

We could avoid this by adding something like the following to `closevideo()`
```
if (videoModalPlayPauseButton === null) return;
```
or we could initialise `videoModalPlayPauseButton` earlier, perhaps when the page has completed loading.

Reply:
The closevideo should not even have been called. Fixed it now.